gwcs/wcs.py

```diff
@@ -545,3 +545,53 @@ def footprint(self, bounding_box=None, center=False):
             vertices = _toindex(vertices)
         result = self.__call__(*vertices, **{'with_bounding_box': False})
         return np.asarray(result)
+
+    # APE 14 implementation
+
+    def pixel_n_dim(self):
+        return self.input_frame.naxes
+
+    def world_n_dim(self):
+        return self.output_frame.naxes
+
+    def pixel_shape(self):
+        return None
+
+    def pixel_bounds(self):
+        return self.bounding_box
+
+    def world_axis_physical_types(self):
+        # Need to match with UCD type
+        return self.output_frame.axes_types
+    
+    def world_axis_units(self):
+        return self.output_frame.unit
+
+    def axis_correlation_matrix(self):
+        return is_separable(self.forward_transform)
+
+    def pixel_to_world_values(self, *pixel_arrays):
+        # need **kwargs
```

Comment:
I am not convinced we do need this here, this API doesn't need to encapsulate the whole of the gWCS functionality.

Having said that I am not sure there is a problem with having it in the function signature. (Other than `output` should be prohibited)

Reply:
There's a suggestion that the result should return NaNs where there's no valid WCS solution. This is controlled by keywords and their current defaults provide this functionality. So yes, currently it's not strictly needed.